app/src/main/java/org/mozilla/fenix/library/history/HistoryAdapter.kt

```diff
@@ -16,13 +16,17 @@ import org.mozilla.fenix.library.history.viewholders.HistoryListItemViewHolder
  */
 class HistoryAdapter(
     private val historyInteractor: HistoryInteractor,
+    private val onEmptyStateChanged: (Boolean) -> Unit,
 ) : PagingDataAdapter<History, HistoryListItemViewHolder>(historyDiffCallback),
     SelectionHolder<History> {
 
     private var mode: HistoryFragmentState.Mode = HistoryFragmentState.Mode.Normal
-    override val selectedItems get() = mode.selectedItems
-    var pendingDeletionIds = emptySet<Long>()
+    private var pendingDeletionItems = emptySet<PendingDeletionHistory>()
     private val itemsWithHeaders: MutableMap<HistoryItemTimeGroup, Int> = mutableMapOf()
+    private var isEmpty = true
```

Comment:
`isEmpty` is a bit generic,  should we add a more descriptive name or add some comments to indicate why we are using it? WDY? :)

Reply:
The only alternative that comes to my mind is `hasItems`, which I find to be even worse :))
Added explanation.

Reply:
👍🏽